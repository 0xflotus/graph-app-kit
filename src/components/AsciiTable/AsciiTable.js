import React, { Component } from "react";
import * as PropTypes from "prop-types";
import asciitable from "ascii-data-table";

export class AsciiTable extends Component {
  constructor(props) {
    super(props);
    this.state = {
      serializedRows: []
    };
  }
  componentDidMount() {
    this.makeState(this.props);
  }
  componentWillReceiveProps(props) {
    this.makeState(props);
  }
  makeState(props) {
    const { data } = props;
    const serializedRows = asciitable.serializeData(data) || [];
    this.setState({ serializedRows });
  }
  render() {
    const { maxColWidth = 70 } = this.props;
    const { serializedRows } = this.state;
    if (!serializedRows.length) return null;
    return (
      <pre>
        {asciitable.tableFromSerializedData(serializedRows, maxColWidth)}
      </pre>
    );
  }
}

AsciiTable.propTypes = {
  data: PropTypes.array,
  maxColWidth: PropTypes.number
};

export default AsciiTable;
