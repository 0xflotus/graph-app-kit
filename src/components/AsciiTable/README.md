Render data in an ascii table for fixed width copy / paste.

```js
const unserializedData = [
  ["Col 1", "Col 2"],
  ["R1C1", ["an", "array"]],
  ["R2C1", { a: "an", b: "object" }],
  ["R3C1", [true, null, false, 1]],
  ["R4C1", "This is a string that will wrap to next line due to maxColWidth"]
];

<AsciiTable data={unserializedData} maxColWidth={25} />
```
