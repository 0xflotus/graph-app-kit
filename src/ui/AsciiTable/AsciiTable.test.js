import React from "react";
import TestRenderer from "react-test-renderer";
import { AsciiTable } from "./AsciiTable";

it("loads AsciiTable without data(no output)", () => {
  const out = TestRenderer.create(<AsciiTable />);
  expect(out.toJSON()).toEqual(null);
});
it("loads AsciiTable data", () => {
  const data = [["h1"], ["r1c1"]];
  const out = TestRenderer.create(<AsciiTable data={data} />);
  expect(out.toJSON().children[0]).toMatchSnapshot();
});
it("re-renders on new data", () => {
  const data1 = [["h1"], ["r1c1"]];
  const data2 = [["h2"], ["r1c1"]];

  const out = TestRenderer.create(<AsciiTable data={data1} />);
  expect(out.toJSON().children[0]).toMatchSnapshot();

  out.update(<AsciiTable data={data2} />);
  expect(out.toJSON().children[0]).toMatchSnapshot();
});
