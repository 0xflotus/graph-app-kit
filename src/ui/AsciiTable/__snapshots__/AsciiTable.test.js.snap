// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads AsciiTable data 1`] = `
"╒══════╕
│\\"h1\\"  │
╞══════╡
│\\"r1c1\\"│
└──────┘"
`;

exports[`re-renders on new data 1`] = `
"╒══════╕
│\\"h1\\"  │
╞══════╡
│\\"r1c1\\"│
└──────┘"
`;

exports[`re-renders on new data 2`] = `
"╒══════╕
│\\"h2\\"  │
╞══════╡
│\\"r1c1\\"│
└──────┘"
`;
